packages/task/src/common/task-protocol.ts

```diff
@@ -159,9 +159,14 @@ export interface TaskConfiguration extends TaskCustomization {
 }
 
 export interface ContributedTaskConfiguration extends TaskConfiguration {
+    /**
+     * ID of the provided/detected task.
+     * This field is not supposed to be used in `tasks.json`.
+     */
+    readonly id: string;
```

Comment:
Hi @RomanNikitenko , does this change look OK to you?

Reply:
Hello, Liang!
Yes, it looks good to me.

If I understand correctly, we need these changes to have more reliable way of matching a task for running.
Right?

Reply:
yes. the label of detected tasks can be customized by the user so it would be easier to use id instead.